solidity/contracts/converter/types/liquidity-pool-v1/LiquidityPoolV1Converter.sol

```diff
@@ -69,6 +81,43 @@ contract LiquidityPoolV1Converter is LiquidityPoolConverter {
         emit Activation(converterType(), anchor, true);
     }
 
+    /**
+      * @dev defines a new reserve token for the converter
+      * can only be called by the owner while the converter is inactive
+      *
+      * @param _token   address of the reserve token
+      * @param _weight  reserve weight, represented in ppm, 1-1000000
+    */
+    function addReserve(IERC20Token _token, uint32 _weight) public override ownerOnly {
+        super.addReserve(_token, _weight);
+
+        isStandardPool =
+            reserveTokens.length == 2 &&
+            reserves[reserveTokens[0]].weight == 500000 &&
+            reserves[reserveTokens[1]].weight == 500000;
+    }
+
+    /**
+      * @dev returns the rate of 1 `_token1` in `_token2` units
+      * note that the rate can only be queried for the reserves
+      *
+      * @param _token1  reserve token to get the rate of one unit of
+      * @param _token2  reserve token to get the rate of one `_token1` unit in
+      *
+      * @return rate (numerator)
+      * @return rate (denominator)
+    */

```

Comment:
This function is used only when the weights are equal, so it is redundant, as we can simply use the balances.
It is also harmful in a way, because it increases the size of N and D, which we need to keep bounded.

Reply:
done